test/Microsoft.AspNetCore.HttpsEnforcement.Tests/HttpsPolicyMiddlewareTests.cs

```diff
@@ -0,0 +1,257 @@
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.HttpsPolicy.Tests
+{
+    public class HttpsPolicyMiddlewareTests
+    {
+        [Fact]
+        public async Task SetOptions_DefaultsSetCorrectly()
+        {
+            var builder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                })
+                .Configure(app =>
+                {
+                    app.UseHttpsPolicy();
+                    app.Run(context =>
+                    {
+                        return context.Response.WriteAsync("Hello world");
+                    });
+                });
+
+            var server = new TestServer(builder);
+            var client = server.CreateClient();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "");
+
+            var response = await client.SendAsync(request);
+
+            // By default, we will not redirect if we only use the Hsts Middleware.
+            Assert.Equal(HttpStatusCode.MovedPermanently, response.StatusCode);
+            Assert.Equal("https://localhost/", response.Headers.Location.ToString());
+        }
+
+        [Theory]
+        [InlineData(301, null, "https://localhost/")]
+        [InlineData(302, null, "https://localhost/")]
+        [InlineData(307, null, "https://localhost/")]
+        [InlineData(308, null, "https://localhost/")]
+        [InlineData(301, 5050, "https://localhost:5050/")]
+        [InlineData(301, 443, "https://localhost/")]
+        public async Task SetOptions_SetStatusCodeTlsPort(int statusCode, int? tlsPort, string expected)
+        {
+
+            var builder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    services.Configure<HttpsPolicyOptions>(options =>
+                    {
+                        options.StatusCode = statusCode;
+                        options.TlsPort = tlsPort;
+                    });
+                })
+                .Configure(app =>
+                {
+                    app.UseHttpsPolicy();
+                    app.Run(context =>
+                    {
+                        return context.Response.WriteAsync("Hello world");
+                    });
+                });
+
+            var server = new TestServer(builder);
+            var client = server.CreateClient();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "");
+
+            var response = await client.SendAsync(request);
+
+            // By default, we will not redirect if we only use the Hsts Middleware.
```

Comment:
This comment is confusing, I'm not sure what you are trying to convey here, but it seems out of context